### ISSUE TITLE
Update 1.21.0 -> 1.21.1 leads to unknown error in renovate

### ISSUE THREAD

**Issue by JochenGuckSnk** [go-gitea/gitea#28252]
### Description

After updating the patch release renovate can not access the issue api anymore. 
The issues endpoint sends a 404.
Downgrading to 1.21.0 resolves the issue.

### Gitea Version

1.21.1

### Can you reproduce the bug on the Gitea demo site?

No

### Screenshots

![grafik](https://github.com/go-gitea/gitea/assets/87813092/a92731b7-87c1-4c15-9af3-34149040c5f0)

### Operating System

Linux/Docker

### How are you running Gitea?

Gitea rootless  docker image.

### Database

PostgreSQL

**Comment by lng2020** [comment]
What's the endpoint and what's the log in your backend?

**Comment by f97** [comment]
I'm also experiencing a similar issue, and I think the reason is that the API cannot access `/api/issues?state=all&type=issues`, but when I remove `&type=issues`, I can access that. I don't know exactly what the issue is, but I have downgraded to version 1.21.0 and am waiting for the next update.

**Comment by JochenGuckSnk** [comment]
Hi @lng2020,

the bot permissions are:
    read:organization
    write:issue
    write:repository
    read:user

the endpoint is: https://{url}/api/v1/repos/{org}/{repo}/issues?state=all&type=issues

I have no backend logs right now but if you need them i can try to reproduce.

**Comment by calvinbui** [comment]
renovate logs:

```
"loggerErrors": [
  {
    "name": "renovate",
    "level": 50,
    "logContext": "<redacted>",
    "repository": "calvinbui/ansible-monorepo",
    "err": {
      "name": "HTTPError",
      "code": "ERR_NON_2XX_3XX_RESPONSE",
      "timings": {
        "start": 1701214916754,
        "socket": 1701214916754,
        "lookup": 1701214916754,
        "connect": 1701214916755,
        "secureConnect": 1701214916756,
        "upload": 1701214916756,
        "response": 1701214916800,
        "end": 1701214916801,
        "phases": {
          "wait": 0,
          "dns": 0,
          "tcp": 1,
          "tls": 1,
          "request": 0,
          "firstByte": 44,
          "download": 1,
          "total": 47
        }
      },
      "message": "Response code 404 (Not Found)",
       "stack": "HTTPError: Response code 404 (Not Found)\n    at Request.<anonymous> (/opt/containerbase/tools/renovate/37.74.1/node_modules/gotdist/source/as-promise/index.js:118:42)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)",
      "options": {
        "headers": {
          "user-agent": "RenovateBot/37.74.1 (https://github.com/renovatebot/renovate)",
          "accept": "application/json",
          "authorization": "***********",
          "accept-encoding": "gzip, deflate, br"
        },
        "url": "https://gitea.<redacted>/api/v1/repos/calvinbui/ansible-monorepo/issues?state=all&type=issues",
        "hostType": "gitea",
        "username": "",
        "password": "",
        "method": "GET",
        "http2": false
      },
      "response": {
        "statusCode": 404,
        "statusMessage": "Not Found",
        "body": {
          "errors": null,
          "message": "The target couldn't be found.",
          "url": "https://gitea.<redacted>/api/swagger"
        },
        "headers": {
          "alt-svc": "h3=\":443\"; ma=2592000",
          "cache-control": "max-age=0, private, must-revalidate, no-transform",
          "content-length": "105",
          "content-type": "application/json;charset=utf-8",
          "date": "Tue, 28 Nov 2023 23:41:56 GMT",
          "strict-transport-security": "max-age=31536000; includeSubDomains; preload",
          "x-content-type-options": "nosniff",
          "x-frame-options": "SAMEORIGIN",
          "connection": "close"
        },
        "httpVersion": "1.1",
        "retryCount": 0
      }
    },
    "msg": "Repository has unknown error"
  }
]
```

**Comment by lv2u** [comment]
Experiencing the same issues as @f97 described. Want to add that the endpoint also returns a 404 when requesting it from the swagger UI, using an access token.

**Comment by lng2020** [comment]
Can't reproduce it. Need some more info.
1. Repo type: private, internal, or public
2. if the user is in a team, what's the team permission?
3. are the user a collaborator of the repo?

**Comment by calvinbui** [comment]
1. Private
2. No team
3. Yes - Administrator

**Comment by JochenGuckSnk** [comment]
1. Private
2. No Team
3. collaborator

the repo is inside a organisation

**Comment by lv2u** [comment]
@lng2020 if you'd like to get access to an environment where this issue occurs, please do let me know. I'm happy to give you access.

**Comment by f97** [comment]
Hi @lng2020, I have tried on try.github.io and gitea.com, and there was no happen. May be it only occurs when set up on Docker. Here is my docker-compose file:

```yml
version: "3"

networks:
  gitea:
    name: gitea

services:
  git:
    image: gitea/gitea
    container_name: git
    environment:
      - USER_UID=1000
      - USER_GID=1000
      - GITEA__actions__ENABLED=true
      - GITEA__repository__DEFAULT_BRANCH=master
      - GITEA__repository.pull-request__DEFAULT_MERGE_STYLE=squash
      - GITEA__repository.pull-request__WORK_IN_PROGRESS_PREFIXES=[WIP]
      - GITEA__mailer__ENABLED=true
      - GITEA__mailer__SMTP_ADDR=smtp.gmail.com
      - GITEA__mailer__SMTP_PORT=465
      - GITEA__mailer__FROM=xxx
      - GITEA__mailer__USER=xxx
      - GITEA__mailer__PASSWD=xxx
    restart: always
    networks:
      - gitea
    volumes:
      - ./:/data
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    ports:
      - 1005:3000
```

**Comment by calvinbui** [comment]
@lng2020 I able to reproduce the 404 by disabling issues on the repo, either by unticking `Enable Repository Issue Tracker` or setting `DISABLED_REPO_UNITS=repo.issues`

**Comment by lv2u** [comment]
Confirming this issue occurs when the built-in issue tracker is disabled.

When enabling the built-in issue tracker again, the issue is resolved.

**Comment by ChriFo** [comment]
> I experienced the same behavior but in my setup the issue module is disabled globally.

This [PR](renovatebot/renovate/pull/26134) included in renovate v37.83.5 fixed it for me.
